test-suite/gradlew

```diff
@@ -0,0 +1,185 @@
+#!/usr/bin/env sh
```

Comment:
> Why do we have a second gradlew here?

Because it's a Gradle [composite build](https://docs.gradle.org/current/userguide/composite_builds.html#composite_builds)

*Why is it a Gradle composite build?*

Because Gradle recommends that plugins be tested this way. See https://docs.gradle.org/current/userguide/testing_gradle_plugins.html#manual-tests

As a side note, the automated tests outlined in that page are a little messy for my tastes. The 'manual' tests look easier to read and maintain compared to the 'automated' ones. I use this method in my other plugin projects.

> I don't think we want to have this as an entirely separate gradle project. 

I agree with you, I don't like it either, but unfortunately that's how Gradle recommends we do it. See above.

Referring to [BUILDING.md](https://github.com/nsoft/uno-jar/blob/master/BUILDING.md), I agree that plugins are difficult to develop and test, even for the ones that don't depend on other modules. What Gradle proposes isn't perfect, but at least Gradle introduced some support for it which doesn't look like a total hack.

> Developer workflow to fix bugs etc should involve running tests regularly, and more particularly the final build before publishing anything to public repositories should include tests being run.

Yes, I fully agree with that. It actually does run the tests. See https://github.com/ngyewch/uno-jar/blob/85fe9576e92de711466f831ad11baf44b62333f9/tests/build.gradle#L247-L250 

```
task testGradle(type: Exec) {
  workingDir = rootProject.file('test-suite')
  commandLine = ["./gradlew", "clean", "build"]
}
```

If the `test-suite` build/test fails, the `testGradle` task should also fail. The test reports in this case can be found under `test-suite`.

Test flow:
* `./gradlew test` -> 
* `:tests:testGradle` -> 
* (`test-suite`) `./gradlew clean build`

Unfortunately, `test-suite` cannot be directly run from main project. It is executed indirectly via an `Exec`.